roles/collect/tasks/main.yml

```diff
@@ -37,39 +37,28 @@
   when:
     - not check_oc.stat.exists
 
-- name: Default OCP insecure login option
+- name: Set facts from OCP config
   set_fact:
-    ocp_insecure_login: '--insecure-skip-tls-verify=false'
+    ocp_validate_cert: '{{ ocp_config["ocp_validate_cert"] }}'
+    ocp_metering_namespace: '{{ ocp_config["ocp_metering_namespace"] }}'
 
-- name: Set OCP insecure login option
+- name: Set facts from OCP config
   set_fact:
-    ocp_insecure_login: '--insecure-skip-tls-verify=true'
-  when:
-    - 'not {{ ocp_config["ocp_validate_cert"] | bool }}'
-
-- name: Define OCP API endpoint without proxy
-  set_fact:
-    collect_endpoint_path: 'api/v1/namespaces/{{ ocp_config["ocp_metering_namespace"] }}'
-  when: not collect_with_proxy
+    ocp_insecure_login: '--insecure-skip-tls-verify={{ ocp_validate_cert }}'
+    collect_endpoint_params: '?name={{ collect_report_name }}&format={{ collect_format }}&namespace={{ ocp_metering_namespace }}'
+    collect_endpoint_path: 'api/v1/namespaces/{{ ocp_metering_namespace }}'
+    collect_endpoint_url: '{{ ocp_config["ocp_api"] }}'
 
-- name: Define OCP API endpoint with proxy
+# conditional tag
+- name: Define facts from OCP config with proxy
   set_fact:
-    collect_endpoint_path: 'api/v1/namespaces/{{ ocp_config["ocp_metering_namespace"] }}/services/https:reporting-operator:http/proxy/api/v1/scheduledreports/get'
-  when: collect_with_proxy
+    collect_endpoint_path: '{{ collect_endpoint_path }}/services/https:reporting-operator:http/proxy/api/v1/scheduledreports/get'
```

Comment:
I think the `scheduledreports` endpoint has gone away and is now just `reports` based on @adberglund  *curl*:

```
curl "http://127.0.0.1:8001/api/v1/namespaces/metering-hccm/services/https:reporting-operator:http/proxy/api/v1/reports/get?name=hccm-openshift-persistentvolumeclaim-lookback&namespace=metering-hccm&format=csv" > hccm_ocp_pvc_lookback_01_31_2019.csv
```